internal/security/setup/option/cache.go

```diff
@@ -69,7 +69,7 @@ func cachePkis() (statusCode exitCode, err error) {
 		if err = doCache(generatedDirPath); err != nil {
 			return exitWithError, err
 		}
-		os.RemoveAll(generatedDirPath)
+		defer os.RemoveAll(generatedDirPath)
```

Comment:
move this `defer` right after the declaration and assignment of `generatedDirPath`